s4e-backend/src/main/java/pl/cyfronet/s4e/controller/GroupController.java

```diff
@@ -0,0 +1,177 @@
+
+    @ApiOperation("Add a new member to the group")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "If member was added"),
+            @ApiResponse(code = 400, message = "Member not added"),
+            @ApiResponse(code = 404, message = "Group or user not found")
+    })
+    @PostMapping("/institutions/{institution}/groups/{group}/members")
+    public ResponseEntity<?> addMember(@PathVariable("group") String groupSlug,
+                                       @RequestBody Long userId,
+                                       @PathVariable("institution") String institutionSlug)
+            throws NotFoundException, GroupUpdateException, BadRequestException {
+        val group = groupService.getGroup(groupSlug)
+                .orElseThrow(() -> new NotFoundException("Group not found for id '" + groupSlug));
+        checkInstitution(institutionSlug, group);
+        val appUser = appUserService.findById(userId)
+                .orElseThrow(() -> new NotFoundException("User not found for id '" + userId));
+        group.addMember(appUser);
+        groupService.update(group);
+        eventPublisher.publishEvent(new OnAddToGroupEvent(appUser, group, LocaleContextHolder.getLocale()));
+        return ResponseEntity.ok().build();
+    }
+
+    @ApiOperation("Remove a member from the group")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "If member was removed"),
+            @ApiResponse(code = 400, message = "Member not removed"),
+            @ApiResponse(code = 404, message = "Group or user not found")
+    })
+    @PostMapping("/institutions/{institution}/groups/{group}/members/{userId}")
+    public ResponseEntity<?> removeMember(@PathVariable("group") String groupSlug,
+                                          @PathVariable Long userId,
+                                          @PathVariable("institution") String institutionSlug)
+            throws NotFoundException, GroupUpdateException, BadRequestException {
+        val group = groupService.getGroup(groupSlug)
+                .orElseThrow(() -> new NotFoundException("Group not found for id '" + groupSlug));
+        checkInstitution(institutionSlug, group);
+        val appUser = appUserService.findById(userId)
+                .orElseThrow(() -> new NotFoundException("User not found for id '" + groupSlug));
+        group.removeMember(appUser);
+        groupService.update(group);
+        eventPublisher.publishEvent(new OnRemoveFromGroupEvent(appUser, group, LocaleContextHolder.getLocale()));
+        return ResponseEntity.ok().build();
+    }
+
+    @ApiOperation("Get a list of groups")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "Successfully retrieved list")
+    })
+    @GetMapping("/institutions/{institution}/groups")
+    public Page<GroupResponse> getAllByInstitutionName(@PathVariable("institution") String institutionSlug) {
+        Page<Group> page = groupService.getAllByInstitution(institutionSlug);
```

Comment:
You have to allow front-end to pass page and limit as parameters (see PlaceController).